packages/p2p/src/components/app.jsx

```diff
@@ -45,13 +63,47 @@ class App extends Component {
         this.setState({ parameters: null });
     };
 
-    setIsAgent = async () => {
-        const agent_info = await requestWS({ p2p_agent_info: 1 });
+    setIsAgent = () =>
+        new Promise(async resolve => {
```

Comment:
You can remove `new Promise` from a function that is `async`. You can simply set `setIsAgent` to `async` and it will resolve with whatever you `return`, like `initializeApp`.